### PR TITLE
XW-3855 | Disable autoplay for AdBlocking users

### PR DESCRIPTION
**Pull request by hakubo**
https://wikia-inc.atlassian.net/browse/XW-3855

### CODE REVIEW

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 1]
```diff
@@ -60,9 +62,46 @@ require([
...
+					setupPlayer(onCreate, false);
+				};
+
+			log('initVideo', 'info', logGroup);
+
+			if (!autoplay) {
+				log('initVideo, autoplay disabled', 'info', logGroup);
+				setupPlayer(onCreate, autoplay);
+			} else {
+				if (!context.opts.pageFairDetection) {
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js:83]: `else if` would reduce the nesting.

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 2]
```diff
@@ -60,9 +62,46 @@ require([
...
+					log('initVideo, pageFairDetection disabled', 'info', logGroup);
+					setupPlayer(onCreate, autoplay);
+				} else {
+					//If pageFair does not respond, start player
+					setTimeout(function () {
+						if (!ooyalaVideoController) {
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js:89]: How about using `clearTimeout()` inside the callbacks instead?

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 3]
```diff
@@ -60,9 +62,46 @@ require([
...
+							log('initVideo, pageFair did not respond under 2000ms, starting video', 'info', logGroup);
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js:90]: Let's put `2000` in a variable, this log might get incorrect easily.

**Review by JamesHawking — APPROVED** [review]

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 4]
```diff
@@ -60,9 +62,46 @@ require([
 			inAutoplayCountries = geo.isProperGeo(instantGlobals.wgArticleVideoAutoplayCountries),
 			inNextVideoAutoplayCountries = geo.isProperGeo(instantGlobals.wgArticleVideoNextVideoAutoplayCountries),
 			autoplayCookieName = 'featuredVideoAutoplay',
-			autoplay = cookies.get(autoplayCookieName) !== '0' && inAutoplayCountries;
+			autoplay = cookies.get(autoplayCookieName) !== '0' && inAutoplayCountries,
+			context = adContext && adContext.getContext(),
+			logGroup = 'FeaturedVideo';
 
 		function initVideo(onCreate) {
+			var adNotBlockedCallback = function () {
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js:70]: I'd add a comment or name this function as `detectAdBlock` so someone looking at this code will know what it does at first glance

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 5]
```diff
@@ -60,9 +62,46 @@ require([
...
+					setupPlayer(onCreate, true);
+				},
+				adBlockedCallback = function () {
+					setupPlayer(onCreate, false);
+				};
+
+			log('initVideo', 'info', logGroup);
+
+			if (!autoplay) {
+				log('initVideo, autoplay disabled', 'info', logGroup);
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js:80]: Why did you all these logs?

**Pull request closed without merging**